R/AbstractPackageReporter.R

```diff
@@ -70,6 +70,9 @@ AbstractPackageReporter <- R6::R6Class(
             
             if (exists("packagePath") && !is.null(packagePath)) {
                 if (dir.exists(packagePath)) {
+                  if (basename(packagePath) != packageName) {
+                    log_fatal(paste0("The folder name at the end of packagePath does not match packageName."))
```

Comment:
I disagree with the way this is written. There is actually no dependence between the name of the folder and the name of the package. The package name in `DESCRIPTION` is the only thing that matters

for example: https://github.com/wesm/feather/tree/master/R

I also don't think we need to guard against this. I don't see this happening often, and don't think it's worth the complexity. I vote that we decline and close this PR